### PR TITLE
Bumping datadog-go (DogStatsD Go client) to 4.2.0

### PR DESCRIPTION
**Pull request by hush-hush**
### What does this PR do?

Bumping datadog-go (DogStatsD Go client) to `4.2.0`

### Motivation

New version brings improvement in performances and fixes.

### CODE REVIEW

File: pkg/trace/metrics/client.go [hunk 1]
```diff
@@ -6,60 +6,12 @@
...
-	Timing(name string, value time.Duration, tags []string, rate float64) error
-	Flush() error
-}
+type StatsClient = statsd.ClientInterface
 
 // Client is a global Statsd client. When a client is configured via Configure,
 // that becomes the new global Statsd client in the package.
 var Client StatsClient = (*statsd.Client)(nil)
-
-// Gauge calls Gauge on the global Client, if set.
```
Comment (hush-hush) [inline pkg/trace/metrics/client.go:27]: This is already done by the main client: https://github.com/DataDog/datadog-go/blob/master/statsd/statsd.go#L463

Every method checks for `nil` so the whole `StatsClient` is useless. I created a type alias line 13 to not change to much code. I'll let @DataDog/agent-apm update the rest of the trace agent to replace `metrics.StatsClient` for `statsd.ClientInterface` if they wan to.

File: pkg/security/module/module.go [hunk 2]
```diff
@@ -202,9 +201,7 @@ func NewModule(cfg *aconfig.AgentConfig) (api.Module, error) {
 	}
 
 	var statsdClient *statsd.Client
-	// statsd segfaults on 386 because of atomic primitive usage with wrong alignment
```
Comment (hush-hush) [inline pkg/security/module/module.go:205]: This has been fixed since 3.5.0

File: pkg/trace/metrics/client.go (hunk 1 above)
Comment (gbbr) [reply pkg/trace/metrics/client.go:27]: Agreed. All this is fine.

Comment (hush-hush) [reply pkg/trace/metrics/client.go]: I'll leave it in the end since it's changing to much code on trace side

**Review by lebauce — APPROVED** [review]

**Review by gbbr — APPROVED** [review]

**Pull request merged**